Feature/ccclxxxvi 3

irynazaiets opened the pull request:
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/CCCLXXXVI-3)


## Code reviewers

- [ ] @Andriy97091
- [ ] @jelysaw
- [ ] @MaksymGromyk
- [ ] @shado1997
- [ ] @tavor118
- [ ] @VolodymyrBodnar
- [ ] @wm139


### Second Level Review

- [ ] @lhalam
- [ ] @alutsysh

## Summary of issue

Implement login/logout functional.
Add base page.
Configure work with static files.

alutsysh commented on moneta/src/python/customauth/forms.py:
```diff
@@ -0,0 +1,19 @@
+        if username and password:
+            user = authenticate(username=username, password=password)
+            if not user:
+                raise forms.ValidationError('This user does not exist')
```
This==?
Use concrete user name or simple "User does not exist"

alutsysh commented on moneta/src/python/customauth/forms.py:
```diff
@@ -0,0 +1,19 @@
+            if not user:
+                raise forms.ValidationError('This user does not exist')
+            if not user.check_password(password):
+                raise forms.ValidationError('Incorrect password!')
```
No !.

irynazaiets replied:
Which algorithm must I use for hashing a password?
Because before checking the password, got from the user with existing password in the database, I must hash got password and after that compare these 2 passwords

alutsysh commented on moneta/src/python/customauth/urls.py:
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+from . import views
```
Why from .?
You should have PYTHONPATH=/PRODROOT/src/python:/PRODROOT/www

alutsysh commented on moneta/src/python/customauth/views.py:
```diff
@@ -0,0 +1,38 @@
+
+
+def login_view(request):
+    # if not request.user.is_authenticated:
```
Do you need this?

alutsysh commented on moneta/src/python/customauth/views.py:
```diff
@@ -0,0 +1,38 @@
+    #     return redirect("/")
+    next = request.GET.get('next')
+    form = UserLoginForm(request.POST)
+    print(form.is_valid())
```
print??? Why not logging?

irynazaiets replied:
It was only for checking the work of my function. Currently, I deleted this

alutsysh commented on moneta/src/python/customauth/views.py:
```diff
@@ -0,0 +1,38 @@
+def login_view(request):
+    # if not request.user.is_authenticated:
+    #     return redirect("/")
+    next = request.GET.get('next')
```
Where do you use it?

irynazaiets replied:
Nowhere. Currently I deleted this

alutsysh commented on moneta/src/python/customauth/views.py:
```diff
@@ -0,0 +1,38 @@
+        else:
+            return redirect('moneta-home')
+
+    print("test")
```
No prints!

irynazaiets replied:
Ok

alutsysh commented on moneta/www/templates/login_app/login.html:
```diff
@@ -0,0 +1,18 @@
+{% block content %}
+<div class="container">
+    <form method="POST">
+        {% csrf_token %}
```
Do you need this?

irynazaiets replied:
Yes, I need

irynazaiets committed:
Implement authentifications

Create auth_user
Create auth_view
Create base templates

 Changes to be committed:
	modified:   db/moneta/CREATE_DB.sql
	modified:   db/moneta/INSERT_DB.sql
	new file:   db/moneta/data/auth_user.sql
	new file:   db/moneta/schema/auth_user.sql
	modified:   manage.py
	deleted:    src/python/db/custom_user.py
	renamed:    www/static/css/__init__.py -> www/customauth/__init__.py
	new file:   www/customauth/models.py
	new file:   www/forms/login_form.py
	modified:   www/settings/settings.py
	modified:   www/settings/urls.py
	new file:   www/static/css/main.css
	new file:   www/templates/base.html
	new file:   www/templates/home.html
	new file:   www/templates/login_app/login.html
	new file:   www/templates/login_app/logout.html
	new file:   www/views/login_view.py

lhalam commented on moneta/db/moneta/schema/auth_user.sql:
```diff
@@ -0,0 +1,12 @@
+/*
```
add this script to CREATE_DB.sql
create data script for this table, and add him in INSERT_DB.sql

irynazaiets replied:
What did you mean?)

lhalam commented on moneta/www/settings/urls.py:
```diff
@@ -14,5 +14,12 @@
+    path('login/', views.login_view, name='login'),
+    path('logout/', views.logout_view, name='logout'),
+    path('auth/', include('customauth.urls')),
+]
```
add endline

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+
+.header a:link, a:visited {
+  color: white;
+}
```
after line add empty line

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+    background-color: #C1E2E3;
+    text-decoration-color: white;
+    flex-basis: 50px;
+
```
remove empty line

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+body{
```
add space before curly braces

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+
```
remove empty lines

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+
+}
+
+.header-label{
```
add space before curly braces

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+}
+
+
+.container a:link, a:visited{
```
add space before curly braces

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+    color: black;
+}
+
+.container{
```
add space before curly braces

lhalam commented on moneta/www/static/css/main.css:
```diff
@@ -0,0 +1,67 @@
+
+}
+
+.header > div
```
move the curly braces from the next line here

lhalam reviewed (commented):
move files to the corresponding directories

alutsysh commented on moneta/www/forms/login_form.py:
```diff
@@ -0,0 +1,35 @@
+
+class UserLoginForm(forms.Form):
+    '''
+
```
empty line?

alutsysh commented on moneta/www/forms/login_form.py:
```diff
@@ -0,0 +1,35 @@
+        if email and password:
+            user = authenticate(email=email, password=password)
+            print(type(user))
+            print(user)
```
no prints! Use logging!

irynazaiets committed:
Merge with dev branch
